root-types.ts

```diff
@@ -0,0 +1,111 @@
+// TODO
+export type Uuid = string & { readonly _: unique symbol };
+
+export interface GameState {
+  game: Game;
+  players: Player[];
+  rounds: Round[];
+}
+
+export type FullGameState = GameState & {
+  piles: Piles;
+};
+
+export type Pack = {
+  abbr: string;
+  name: string;
+  icon?: string;
+};
+
+export enum GameStatus {
+  Created = 'created',
+  Running = 'running',
+  Ended = 'ended',
+}
+
+export interface Game {
+  id: Uuid;
+  requiresPassword: boolean;
+  status: GameStatus;
+  packs: Pack[];
+  timeouts: {
+    playing: number;
+    revealing: number;
+    judging: number;
+    betweenRounds: number;
+  };
+  rules: {
+    handSize: number;
+    winnerPoints: number;
+    randoCardrissianPlayers: number;
```

Comment:
Could `rules` contain something like `Mode[]` to be more flexibel?

Reply:
Fair point, see update.